src/main/java/de/mediathekview/mserver/crawler/zdf/tasks/ZdfFilmDetailTask.java

```diff
@@ -0,0 +1,105 @@
+package de.mediathekview.mserver.crawler.zdf.tasks;
+
+import com.google.gson.reflect.TypeToken;
+import de.mediathekview.mlib.daten.Film;
+import de.mediathekview.mlib.daten.GeoLocations;
+import de.mediathekview.mlib.daten.Resolution;
+import de.mediathekview.mserver.crawler.basic.AbstractCrawler;
+import de.mediathekview.mserver.crawler.basic.AbstractRecrusivConverterTask;
+import de.mediathekview.mserver.crawler.zdf.ZdfEntryDto;
+import de.mediathekview.mserver.crawler.zdf.ZdfVideoUrlOptimizer;
+import de.mediathekview.mserver.crawler.zdf.json.DownloadDto;
+import de.mediathekview.mserver.crawler.zdf.json.ZdfDownloadDtoDeserializer;
+import de.mediathekview.mserver.crawler.zdf.json.ZdfFilmDetailDeserializer;
+import java.lang.reflect.Type;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import javax.ws.rs.client.WebTarget;
+import mServer.crawler.CrawlerTool;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ZdfFilmDetailTask extends ZdfTaskBase<Film, ZdfEntryDto> {
+
+  private static final Logger LOG = LogManager.getLogger(ZdfFilmDetailTask.class);
+
+  private static final Type OPTIONAL_FILM_TYPE_TOKEN = new TypeToken<Optional<Film>>() {
+  }.getType();
+  private static final Type OPTIONAL_DOWNLOAD_DTO_TYPE_TOKEN = new TypeToken<Optional<DownloadDto>>() {
+  }.getType();
+
+  private final ZdfVideoUrlOptimizer optimizer = new ZdfVideoUrlOptimizer();
+
+  public ZdfFilmDetailTask(AbstractCrawler aCrawler,
+      ConcurrentLinkedQueue<ZdfEntryDto> aUrlToCrawlDtos, Optional<String> aAuthKey) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDtos" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)